proposals/interface-types/working-notes/Instructions.md

```diff
@@ -0,0 +1,252 @@
+# Interface instructions
+
+This proposal defines new instructions that operate over interface value types. These instructions have the same form as core instructions; they push and pop values on an execution stack and support single pass validation.
+
+The new interface instructions are summarized below.
+
+```
+#interface-instr ::=
+  | bool.lift_iMM
+  | bool.lower_iMM
+
+  | sNN.lift_iMM
+  | sNN.lower_iMM
+
+  | uNN.lift_iMM
+  | uNN.lower_iMM
+
+  | string.lift_memory ..
+  | string.lower_memory ..
+
+  | record.lift ..
+  | record.lower ..
+
+  | array.lift_memory ..
+  | array.lower_memory ..
+
+  | invoke ..
```

Comment:
we need several invocation instructions:

call-export (both for core wasm and of IT functions)
invoke-method

Reply:
We will need invoke. 
There is a difference between call and call-export. It has to do with function tables (not being available everywhere)

Reply:
I've included `call` and `call_indirect` to the list of core wasm instructions we're importing into this document. What's the reason we need a new `call-export`?

Reply:
I think we should hold off adding `invoke-method` until we add `protocol`s.  I think it's fine to have `call-export` for the time being because, as Interface Types is currently framed, it's fundamentally different than `call` because it takes an export string, not a function index.  Once we can rebase on linking, then I think everything can just be `call`s.